Circular dependency between sys/util.h and sys/time_units.h

**Describe the bug**

The `WAIT_FOR` macro in `zephyr/sys/util.h` uses `k_us_to_cyc_ceil32` https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/util.h#L616

This function is defined in `zephyr/sys/time_units.h`, but `time_units.h` itself includes `zephyr/sys/util.h` https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/time_units.h#L11 , making a circular dependency

**How to reproduce**

Create a C file that uses `WAIT_FOR` e.g.

```
#include <zephyr/sys/util.h>
...
bool my_func() { return false; }
...
WAIT_FOR(my_func(), 1000, NULL);
```

This will fail to compile with something like

```
.../zephyr/include/zephyr/sys/util.h:551:26: error: implicit declaration of function 'k_us_to_cyc_ceil32' [-Werror=implicit-function-declaration]
  551 |   uint32_t cycle_count = k_us_to_cyc_ceil32(timeout); \
```

**Expected behavior**

Headers should be self-contained so that users can include `zephyr/sys/util.h` without knowing that they also need to include `zephyr/sys/time_units.h` with a particular include ordering.

**Impact**

Users trying to get access to `WAIT_FOR` from `util.h` will hit a confusing compiler error around `k_us_to_cyc_ceil32` if they didn't happen to transitively include `time_units.h` already.

--- a/include/zephyr/sys/util.h
+++ b/include/zephyr/sys/util.h
@@ -559,6 +559,11 @@ char *utf8_lcpy(char *dst, const char *src, size_t n);
 }
 #endif
 
+/* This file must be included at the end of the !_ASMLANGUAGE guard.
+ * It depends on macros defined in this file above which cannot be forward declared.
+ */
+#include <zephyr/sys/time_units.h>
+
 #endif /* !_ASMLANGUAGE */
 
 /** @brief Number of bytes in @p x kibibytes */

